src/dsm/headers/Graph.hpp

```diff
@@ -778,10 +794,10 @@ namespace dsm {
         if (visitedNodes.find(neighbour.first) != visitedNodes.end()) {
           continue;
         }
-        double streetLength = (*(this->street(source, neighbour.first)))->length();
+        double streetWeight = getStreetWeight(this, source, neighbour.first);
```

Comment:
Is `double` type a fair assumption?
We could make it more general... but effectively sounds like more effort than utility.